packages/node/src/transports/base.ts

```diff
@@ -62,6 +65,9 @@ export abstract class BaseTransport implements Transport {
   /** The Agent used for corresponding transport */
   public client?: http.Agent | https.Agent;
 
+  /** The function used to parse URLs */
+  public urlParser?: UrlParser;
```

Comment:
Can we actually make this non-optional, and move the default `utl => new URL(url)` here? Otherwise it may be a breaking change for people using custom transports.

Reply:
@kamilogorek before I jump into this rabbit hole, is this what you mean?
```diff
diff --git packages/node/src/transports/base.ts packages/node/src/transports/base.ts
index 1b4977de..3413cbb8 100644
--- packages/node/src/transports/base.ts
+++ packages/node/src/transports/base.ts
@@ -65,9 +65,6 @@ export abstract class BaseTransport implements Transport {
   /** The Agent used for corresponding transport */
   public client?: http.Agent | https.Agent;
 
-  /** The function used to parse URLs */
-  public urlParser?: UrlParser;
-
   /** API object */
   protected _api: API;
 
@@ -82,6 +79,9 @@ export abstract class BaseTransport implements Transport {
     this._api = new API(options.dsn, options._metadata);
   }
 
+  /** Default function used to parse URLs */
+  public urlParser: UrlParser = url => new URL(url);
+
   /**
    * @inheritDoc
    */
@@ -230,9 +230,6 @@ export abstract class BaseTransport implements Transport {
         if (!this.module) {
           throw new SentryError('No module available');
         }
-        if (!this.urlParser) {
-          throw new SentryError('No URL parser configured');
-        }
         const options = this._getRequestOptions(this.urlParser(sentryReq.url));
         const req = this.module.request(options, (res: http.IncomingMessage) => {
           const statusCode = res.statusCode || 500;
```

I also feel like URL parser can belong as an (optional) option in TransportOptions. What do you think?

Reply:
Yes, exactly that. Tbh I don't see anyone ever wanting to override this other than use cases like yours, where completely custom transport is required.

Reply:
@kamilogorek gotcha, done.